### ISSUE TITLE
compilation error on fresh installed Arch Linux

### ISSUE THREAD

**Issue by pojienie** [synfig/synfig#2258]
when running 2-build-debug.sh, I encountered an error:
```
checking for mlt++... configure: error:  ** You need to install mlt++
```

After resolving the above issue, I also encountered the following error:
```
In file included from synfig/synfig-core/src/tool/optionsprocessor.cpp:54:
synfig/synfig-core/src/tool/optionsprocessor.h:130:38: error: ISO C++17 does not allow dynamic exception specifications
  130 |         void process_debug_options() throw (SynfigToolException&);
      |                                      ^~~~~
synfig/synfig-core/src/tool/optionsprocessor.cpp:845:55: error: ISO C++17 does not allow dynamic exception specifications
  845 | void SynfigCommandLineParser::process_debug_options() throw (SynfigToolException&)
      |                                                       ^~~~~
make[3]: *** [Makefile:625: synfig-optionsprocessor.o] Error 1
```
**To Reproduce**
right after setting up Arch Linux
git clone https://github.com/synfig/synfig
cd synfig
./1-setup-linux-native.sh
./2-build-debug.sh

**Expected behavior**
build to completion without error.

**System information:**
 - OS: Arch Linux
 - Commit ID: 798bfbf95983ff6c01ca019ac72b390fbc259eab

**Additional comments**
For the first problem, it can be easily fixed after installing mlt6 (pacman -S mlt6).
The synfig repository in the of [Community repository](https://archlinux.org/packages/community/x86_64/synfig/) has `mlt6` dependency instead of `mlt`. So I believe changing the `mlt` to `mlt6` in `1-setup-linux-native.sh` is enough to fix this problem.
Not sure how I should do about the second issue, though.

**Comment by rodolforg** [comment]
Currently we use C++11 for synfig cuilding. This is already a (deprecation) warning on C++11.
We should  fix it…

For now, try to remove ` throw (SynfigToolException&)`...

**Comment by pojienie** [comment]
Yes, removing them works for me! Thank you :)

I studied further and found out that the `throw (SynfigToolException)` in the function declaration basically tells the programmer that this function might throw the specified exception so removing them doesn't seem to matter at all. It was deprecated in favor of `noexcept(true)` judging from what I read [here](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0003r5.html).

If it's okay, I can submit a pull request that changes `mlt` to `mlt6` for `ID_LIKE=arch` and remove those `throw (SynfigToolException)`

**Comment by rodolforg** [comment]
Fixed on master PR #2335
